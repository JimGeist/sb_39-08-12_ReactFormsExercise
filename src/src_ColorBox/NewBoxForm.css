label {
    margin: 10px 5px 10px 5px;
}

input {
    margin-right: 20px; 
    font-weight: bold;
}

.NewBoxForm-BtnAdd {
    display: block;
    padding: 3px 10px 3px 10px;
    margin: 15px 0 15px 5px;
}

.NewBoxForm {
    background-color: cyan;
    margin: 5px 0 5px 5px;
    border-radius: 5px;
    border-style: solid;
    border-color:darkblue;
    border-width: 3px;
}

.NewBoxForm-InpNbr {
    width: 4.5rem;
}

