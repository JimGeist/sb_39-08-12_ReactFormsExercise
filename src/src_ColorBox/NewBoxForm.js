import React, { useState } from "react";

import './NewBoxForm.css';

function NewBoxForm({ addBoxFx }) {

    const INITIAL_VALUES = { width: '', height: '', bgColor: '' };

    const [formData, setFormData] = useState(INITIAL_VALUES);

    const handleChange = (evt) => {

        const { name, value } = evt.target;
        setFormData(formData => (
            { ...formData, [name]: value }
        ));

    };

    const handleSubmit = (evt) => {

        evt.preventDefault();
        addBoxFx({ ...formData, id: Date.now() });
        setFormData(INITIAL_VALUES);

    };

    return (
        <form className="NewBoxForm" onSubmit={handleSubmit}>
            <label htmlFor="width">Box Width:</label>
            <input
                id="width"
                className="NewBoxForm-InpNbr"
                type="number"
                name="width"
                value={formData.width}
                onChange={handleChange} />
            <label htmlFor="height">Box Height:</label>
            <input
                id="height"
                className="NewBoxForm-InpNbr"
                type="number"
                name="height"
                value={formData.height}
                onChange={handleChange} />
            <label htmlFor="bgColor">Box Color:</label>
            <input
                id="bgColor"
                type="text"
                name="bgColor"
                value={formData.bgColor}
                onChange={handleChange} />
            <button className="NewBoxForm-BtnAdd">Add Box</button>
        </form>
    );
}

export default NewBoxForm;
