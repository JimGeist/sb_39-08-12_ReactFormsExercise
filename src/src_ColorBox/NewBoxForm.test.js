import React from "react";
import { render, screen } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

// Smoke Test -- does NewBoxFormList render?
it("renders the NewBoxForm component without crashing", () => {
    render(<NewBoxForm />);
});

// NewBoxFormList snapshot tests
it("NewBoxForm component matches snapshot", () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("has a width, height, and color labels and an Add Box button", function () {
    const { getByLabelText, getByText } = render(<NewBoxForm />);

    // expect 'Box Width:', 'Box Height:', and 'Box Color:' labels
    expect(getByLabelText("Box Width:")).toBeInTheDocument();
    expect(getByLabelText("Box Height:")).toBeInTheDocument();
    expect(getByLabelText("Box Color:")).toBeInTheDocument();

    // expect 'Add Box' button
    expect(getByText("Add Box")).toBeInTheDocument();

});