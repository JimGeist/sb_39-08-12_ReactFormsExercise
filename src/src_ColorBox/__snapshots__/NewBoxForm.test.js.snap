// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBoxForm component matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      for="width"
    >
      Box Width:
    </label>
    <input
      class="NewBoxForm-InpNbr"
      id="width"
      name="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Box Height:
    </label>
    <input
      class="NewBoxForm-InpNbr"
      id="height"
      name="height"
      type="number"
      value=""
    />
    <label
      for="bgColor"
    >
      Box Color:
    </label>
    <input
      id="bgColor"
      name="bgColor"
      type="text"
      value=""
    />
    <button
      class="NewBoxForm-BtnAdd"
    >
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
